portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,82 @@
 
 package com.google.sps.servlets;
 
+import java.lang.StringBuffer;
 import java.io.IOException;
+import java.time.format.DateTimeFormatter;  
+import java.time.LocalDateTime;   
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private ArrayList<Comment> allComments = new ArrayList<Comment>();
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    /**
+    * Class to organize comments for more detailed displays. By creating 
+    * another class, adding new parameters and information to keep track 
+    * of may be easier.
+    */
+    private static class Comment {
+        private String name;
+        private String comment;
+        private String date;
+
+        Comment(String newName, String newComment) {
+            name = newName;
+            comment = newComment;
+
+            DateTimeFormatter dtf = DateTimeFormatter.ofPattern("LLL dd, yyyy HH:mm:ss");  
+            LocalDateTime now = LocalDateTime.now();  
+            date = dtf.format(now);  
+        }
+
+        /** @return a formatted string with all of the comment information. */
+        public String toString() {
+            return name + ": " + comment + " (" + date +")";
+        }
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String name = request.getParameter("name");
+        String comment = request.getParameter("comment");
+
+        Comment entry = new Comment(name, comment);
+
+        allComments.add(entry);
+        response.sendRedirect("/index.html");
+        return;
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String allCommentsAsStrings = getCommentsAsStrings();
```

Comment:
consider the division of responsibilities - what fits better on the server side vs client side? One way of looking at it is that the server is responsible for providing the data, and client is responsible for deciding how to display the data... This will be addressed in further code labs, so I am not particularly concerned about it now, but a very important way to organize code/data is via model (data) vs view (how the data is displayed) vs controller (how the data is manipulated). Is it possible to organize this such that the server (this servlet) simply provides the data (without specifying how it needs to be displayed), and the client (your HTML/CSS/JS) decides how to display it?